.github/CONTRIBUTING.md

```diff
@@ -1,20 +1,46 @@
-# Documentation
+# Contributing
 
-We use a fork and pull model, you can read about
-[forking repos at github help](https://help.github.com/articles/fork-a-repo/).
+We use [GitHub](https://github.com/ElektraInitiative/libelektra/) to maintain this project. First of all you will need a GitHub account and [Git](https://www.git-scm.com/).
 
-Any typo fixes, spelling fixes, fixes in documentation,
-or README.md can be done without any further considerations!
-Simply do the fixes in your cloned repo and create a pull request.
+## Configure Git
 
-# Issues
+```sh
+$ git config --global user.name "Your Name"
+$ git config --global user.email "your_email@example.com"
+```
 
-If you have any questions you can ask them using
-[github issue](https://github.com/ElektraInitiative/libelektra/issues/new).
-Simply ignore the template in this case.
+When you connect to a GitHub repository from Git, you have to set up the [authentication](https://help.github.com/en/articles/set-up-git#next-steps-authenticating-with-github-from-git) settings.
 
-See also [FAQ](/doc/help/elektra-faq.md) for some details
-about creating issues.
+## Fork & Sync
+
+We use a fork/sync and pull-request (see below) model at GitHub, follow this short [tutorial](https://help.github.com/articles/fork-a-repo/) to get familiar with forking and syncing.
+
+Sync the fork of the repository to keep it up-to-date with the upstream repository by using the following git commands in your local repository. Important: Using these commands will avoid unnecessary merge commits while you are working on a pull-request (see below):
+
+```sh
+$ git fetch upstream
+$ git rebase upstream/master
+$ git push origin master --force
+```
+
+## Issues
+
+Check the [Ideas](/doc/IDEAS.md) page if you are searching for a good topic to start with. You can also visit the [issue-tracker](https://github.com/ElektraInitiative/libelektra/issues). Do not hesitate to open a new issue if you want to ask a question, report or fix a bug or approach new topics.
+
+## Creating a Pull-Request
+
+If you want to publish your local changes to this project you have to create a new pull-request.
+
+1. Open GitHub and navigate to _your_ libeketra-fork
+2. In the tab _Codes_ press the _New Pull-Request_ button and choose a title and fill in the [pull-request template](/.github/PULL_REQUEST_TEMPLATE.md)
+3. Now you should be able to set up the _Comparing changes_ settings. Choose the head repository and base repository. E.g.: if you have some changes in the master branch of your forked repository, select *head repository: yournickname/libelektra and *compare: master* and *base repositroy: ElektraInitiative/libelektra* and *base: master\*
+4. Add some information about the changes in the release notes (path of the file: /libelektra/doc/news/\_preparation_next_release.md), skipping this step may cause a rejected pull-request
```

Comment:
to avoid frustration of the developer you should write about the formatting. The build server instantly fails if there is just have a single space too much at the end of the line (scripts\dev\reformat-all). This might require the contributor to either install docker (doc\tutorials\run_reformatting_script_with_docker.md) or the scripts itself but with additional tools installed (written in here doc\CODING.md).